### ISSUE TITLE
Boolean attribute doesn't behave as might by expected for web component

### ISSUE THREAD

**Issue by jrencz** [bitovi/react-to-web-component#162]
Hi!

I noticed that the chosen behaviour for boolean transform is 

https://github.com/bitovi/react-to-web-component/blob/1a6886a221e9f23710c7ba958d6d260bf8f7dc43/packages/core/src/transforms/boolean.ts#L3-L6

which is:

> it's `true` if value of attribute starts with `t`, `y` or a non-0 number.

and it's tested:

https://github.com/bitovi/react-to-web-component/blob/002a007c40fa67fc95c9cddbfde335b22a949ed8/tests/react-to-webcomponent.test.jsx#L352-L353

On the other hand HTML boolean (https://developer.mozilla.org/en-US/docs/Glossary/Boolean/HTML) attribute is considered `true` just because it's present:

> If an HTML tag contains a boolean attribute - no matter the value of that attribute - the attribute is set to `true` on that element. If an HTML tag does not contain the attribute, the attribute is set to `false`.

That leads to interesting behaviour - if attribute is present, but not assigned a value (i.e. it's HTML Boolean), it's cast to `false` (not just `undefined` but actually the polar opposite)

I detected it wile using with Lit booleans (https://lit.dev/docs/templates/expressions/#boolean-attribute-expressions) - Lit implements HTML boolean.

---

I think it can be resolved in a backwards compatible way. 

I see that transforms are given the attribute name as 2nd argument

- in constructor
  https://github.com/bitovi/react-to-web-component/blob/002a007c40fa67fc95c9cddbfde335b22a949ed8/packages/core/src/core.ts#L100
- in `attributeChangedCallback`
  https://github.com/bitovi/react-to-web-component/blob/002a007c40fa67fc95c9cddbfde335b22a949ed8/packages/core/src/core.ts#L130

So parse could be made into

```ts
  parse: (value, attribute) => 
    // Check if it's HTML boolean
    value === undefined
    || value === ''
    || value === attribute
    /// Check if it's boolean by value
    || /^[ty1-9]/i.test(value), 
```

I don't have an idea for `stringify`, though. It would have to be a choice of the author (i.e. differently named transform) I guess

**Comment by justinbmeyer** [comment]
fwiw, I agree we should match the default behavior of the DOM ... attribute existence is `true`

**Comment by jrencz** [comment]
-_-'

My bad - I didn't look through the open PRs: there's #123 addressing the same issue I raised, yet in a different way than I proposed (absence of value is checked not at the level of `Transform['parse']` as I suggested, but before it's called)

Some of the ideas from #123 (namely - the access to attribute name in `Transform['parse']`) were introduced by #146, but besides it also has a proposed deprecation

Anyway: I created #164 that aims to resolve this issue

**Comment by christopherjbaker** [comment]
I agree that it should match the DOM behaviour very closely, but we didn't realize this until after the 2.0 release and modifying it now would be a backwards-breaking change.

The most confusing parts to me of DOM behaviour is that `checked=""` and `checked="false"` are both treated as true. Those would be the two exceptions I might argue for, though I'm not sure how much we should break the behaviour without releasing 3.0.

**Comment by jrencz** [comment]
In #164 `"false"` is preserved a `true` but making `""` a true contradicts the behavior compatible with Lit https://lit.dev/docs/templates/expressions/#boolean-attribute-expressions getting to which is my primary aim

Maybe then, for 2.x there could be a second `dom_boolean` which would allow the author to decide how the custom element behaves?
